crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -293,6 +314,68 @@ impl Timeline {
         }
     }
 
+    /// Send a reply to the given event.
+    ///
+    /// Currently it only supports events with an event ID and JSON being
+    /// available (which can be removed by local redactions). This is subject to
+    /// change. Please check [`EventTimelineItem::can_be_replied_to`] to decide
+    /// whether to render a reply button.
+    ///
+    /// The sender of `reply_item` will be added to the mentions of the reply if
+    /// and only if `reply_item` has not been written by the sender.
+    ///
+    /// # Arguments
+    ///
+    /// * `content` - The content of the reply
+    ///
+    /// * `event_id` - The event id of the item you want to reply to
+    ///
+    /// * `forward_thread` - Usually `Yes`, unless you explicitly want to the
+    ///   reply to show up in the main timeline even though the `reply_item` is
+    ///   part of a thread
+    pub async fn send_reply_with_event(
+        &self,
+        content: RoomMessageEventContentWithoutRelation,
+        event_id: &EventId,
+        forward_thread: ForwardThread,
+    ) -> Result<(), UnsupportedReplyItem> {
+        if let Some(timeline_item) = self.item_by_event_id(event_id).await {
+            return self
+                .send_reply_with_event_timeline_item(content, &timeline_item, forward_thread)
+                .await;
+        }
+
+        let Ok(event) = self.room().event(event_id).await else {
+            return Err(UnsupportedReplyItem::MISSING_EVENT);
+        };
+
+        let raw_sync_event: Raw<AnySyncTimelineEvent> = event.event.cast();
+        let Ok(sync_event) = raw_sync_event.deserialize_as::<AnySyncTimelineEvent>() else {
+            return Err(UnsupportedReplyItem::DESERIALIZATION_ERROR);
+        };
+
+        // Likely needs to be changed or renamed?
+        let Some(timeline_item_content) =
+            TimelineItemContent::from_latest_event_content(sync_event.clone())
```

Comment:
I think we can reuse this function, but maybe is worth changing it's name to `from_any_sync_event` ?

Reply:
@bnjbvr what about this?